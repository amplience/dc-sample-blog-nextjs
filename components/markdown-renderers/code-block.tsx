import React, { ReactElement } from 'react';
import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { atomDark } from 'react-syntax-highlighter/dist/cjs/styles/prism';

export const MarkdownCodeBlock = ({ language, value }: { language: string; value: string }): ReactElement => {
  if (value === undefined) {
    value = '';
  }

  return (
    <SyntaxHighlighter language={language} wrapLines={true} showLineNumbers={true} style={atomDark}>
      {value}
    </SyntaxHighlighter>
  );
};
