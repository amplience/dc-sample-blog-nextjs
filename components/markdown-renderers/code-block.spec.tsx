import React from 'react';
import renderer from 'react-test-renderer';
import { MarkdownCodeBlock } from './code-block';

describe('CodeBlock', (): void => {
  it('should render some code as a pre tag with syntax highlighting', (): void => {
    const code: { language: string; value: string } = {
      language: 'javascript',
      value: "```\nimport { something } from 'somewhere';\n\nconst someThing = something();\n```"
    };

    const wrapper = renderer.create(<MarkdownCodeBlock language={code.language} value={code.value} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render empty line if value is undefined', (): void => {
    const code: { language: string; value?: string } = {
      language: 'javascript',
      value: undefined
    };

    const wrapper = renderer
      .create(<MarkdownCodeBlock language={code.language} value={code.value as string} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
