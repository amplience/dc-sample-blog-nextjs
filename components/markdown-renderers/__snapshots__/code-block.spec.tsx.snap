// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock should render empty line if value is undefined 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "4",
      "OTabSize": "4",
      "WebkitHyphens": "none",
      "background": "#1d1f21",
      "borderRadius": "0.3em",
      "color": "#c5c8c6",
      "direction": "ltr",
      "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
      "hyphens": "none",
      "lineHeight": "1.5",
      "margin": ".5em 0",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "1em",
      "tabSize": "4",
      "textAlign": "left",
      "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
      "whiteSpace": "pre",
      "wordBreak": "normal",
      "wordSpacing": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "color": "#c5c8c6",
        "direction": "ltr",
        "float": "left",
        "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "paddingRight": "10px",
        "tabSize": "4",
        "textAlign": "left",
        "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
      }
    }
  >
    <span
      className="react-syntax-highlighter-line-number"
    >
      1

    </span>
  </code>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "color": "#c5c8c6",
        "direction": "ltr",
        "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "tabSize": "4",
        "textAlign": "left",
        "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
      }
    }
  />
</pre>
`;

exports[`CodeBlock should render some code as a pre tag with syntax highlighting 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "4",
      "OTabSize": "4",
      "WebkitHyphens": "none",
      "background": "#1d1f21",
      "borderRadius": "0.3em",
      "color": "#c5c8c6",
      "direction": "ltr",
      "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
      "hyphens": "none",
      "lineHeight": "1.5",
      "margin": ".5em 0",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "1em",
      "tabSize": "4",
      "textAlign": "left",
      "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
      "whiteSpace": "pre",
      "wordBreak": "normal",
      "wordSpacing": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "color": "#c5c8c6",
        "direction": "ltr",
        "float": "left",
        "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "paddingRight": "10px",
        "tabSize": "4",
        "textAlign": "left",
        "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
      }
    }
  >
    <span
      className="react-syntax-highlighter-line-number"
    >
      1

    </span>
    <span
      className="react-syntax-highlighter-line-number"
    >
      2

    </span>
    <span
      className="react-syntax-highlighter-line-number"
    >
      3

    </span>
    <span
      className="react-syntax-highlighter-line-number"
    >
      4

    </span>
    <span
      className="react-syntax-highlighter-line-number"
    >
      5

    </span>
  </code>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "color": "#c5c8c6",
        "direction": "ltr",
        "fontFamily": "Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "tabSize": "4",
        "textAlign": "left",
        "textShadow": "0 1px rgba(0, 0, 0, 0.3)",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
      }
    }
  >
    <span
      className="token template-string token token template-string token token template-string token"
      style={
        Object {
          "color": "#A8FF60",
        }
      }
    >
      <span
        className="token template-string token template-punctuation"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        \`
      </span>
      <span
        className="token template-string token"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        

      </span>
    </span>
    <span
      className="token template-string token token template-string token token template-string token"
      style={
        Object {
          "color": "#A8FF60",
        }
      }
    >
      import { something } from 'somewhere';

    </span>
    <span
      className="token template-string token token template-string token token template-string token"
      style={
        Object {
          "color": "#A8FF60",
        }
      }
    >
      

    </span>
    <span
      className="token template-string token token template-string token token template-string token"
      style={
        Object {
          "color": "#A8FF60",
        }
      }
    >
      const someThing = something();

    </span>
    <span
      className="token template-string token token template-string token token template-string token"
      style={
        Object {
          "color": "#A8FF60",
        }
      }
    >
      <span
        className="token template-string token"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        
      </span>
      <span
        className="token template-string token template-punctuation"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        \`
      </span>
      <span
        className="token template-string token template-punctuation"
        style={
          Object {
            "color": "#A8FF60",
          }
        }
      >
        \`
      </span>
    </span>
  </code>
</pre>
`;
